File: 240704/9-1 정사각형/9-1-square.js
const fs = require('fs');
const n = Number(fs.readFileSync(0).toString().trim());

const nums = new Array(9).fill().map((_,i) => i+1).reverse();

let cnt = 0;

for(let i = 0; i < n; i++){
    let str = "";
    for(let j = 0; j < n; j++){
        str += nums[cnt % 9];
        cnt++;
    }
    console.log(str);
}

File: 240704/9-1 정사각형/9-1-square.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        int[] nums = new int[9];

        for(int i = 9; i > 0; i--){
            nums[9-i] = i;
        }


        int cnt = 0;

        for(int i = 0; i < n; i++){
            for(int j = 0; j < n; j++){
                System.out.print(nums[cnt % 9]);
                cnt++;
            }
            System.out.println();
        }
    }
    
}